cabot/cabotapp/run_window.py

```diff
@@ -0,0 +1,96 @@
+import json
+from datetime import time
+from dateutil import rrule
+from django.utils import timezone
+
+
+class CheckRunWindow:
+    class Window:
+        start_time = None  # type: time
+        end_time = None  # type: time
+        recurrence = None  # type: rrule.rrule
+
+        def __init__(self, start_time, end_time, recurrence):
+            # type: (time, time, rrule.rrule) -> None
+            assert isinstance(start_time, time)
+            assert isinstance(end_time, time)
+            assert isinstance(recurrence, rrule.rrule)
```

Comment:
I don't love asserts in code vs throwing exceptions but not a huge deal

Reply:
Eh, it was just to stop me from accidentally passing in a `datetime` instead of a `time` (which I did by mistake a few times).